isis/src/base/objs/SpicePosition/SpicePosition.cpp

```diff
@@ -1323,78 +1332,21 @@ namespace Isis {
    *   @history 2009-08-03 Jeannie Walldren - Original version
    */
   void SpicePosition::SetEphemerisTimeHermiteCache() {
+    if (p_hasVelocity) {
+      ale::State state = m_state->getState(p_et, ale::SPLINE);
 
-    // what if framing camera???
+      p_coordinate[0] = state.position.x;
+      p_coordinate[1] = state.position.y;
+      p_coordinate[2] = state.position.z;      
 
-    // On the first SetEphemerisTime, create our splines. Later calls should
-    // reuse these splines.
-    if(p_xhermite == NULL) {
-      p_overrideTimeScale = 1.;
-      p_override = ScaleOnly;
-      ComputeBaseTime();
-
-      p_xhermite = new NumericalApproximation(
-          NumericalApproximation::CubicHermite);
-      p_yhermite = new NumericalApproximation(
-          NumericalApproximation::CubicHermite);
-      p_zhermite = new NumericalApproximation(
-          NumericalApproximation::CubicHermite);
-
-      vector<double> xvalues(p_cache.size());
-      vector<double> xhermiteYValues(p_cache.size());
-      vector<double> yhermiteYValues(p_cache.size());
-      vector<double> zhermiteYValues(p_cache.size());
-
-      vector<double> xhermiteVelocities(p_cache.size());
-      vector<double> yhermiteVelocities(p_cache.size());
-      vector<double> zhermiteVelocities(p_cache.size());
-
-      for(unsigned int i = 0; i < p_cache.size(); i++) {
-        vector<double> &cache = p_cache[i];
-        double &cacheTime = p_cacheTime[i];
-        xvalues[i] = (cacheTime - p_baseTime) / p_timeScale;
-
-        xhermiteYValues[i] = cache[0];
-        yhermiteYValues[i] = cache[1];
-        zhermiteYValues[i] = cache[2];
-
-        if(p_hasVelocity) {  // Line scan camera
-          vector<double> &cacheVelocity = p_cacheVelocity[i];
-          xhermiteVelocities[i] = cacheVelocity[0];
-          yhermiteVelocities[i] = cacheVelocity[1];
-          zhermiteVelocities[i] = cacheVelocity[2];
-        }
-        else { // Not line scan camera
-          throw IException(IException::Io, "No velocities available.",
-                           _FILEINFO_);
-        }
-      }
-
-      p_xhermite->AddData(xvalues, xhermiteYValues);
-      p_xhermite->AddCubicHermiteDeriv(xhermiteVelocities);
-
-      p_yhermite->AddData(xvalues, yhermiteYValues);
-      p_yhermite->AddCubicHermiteDeriv(yhermiteVelocities);
-
-      p_zhermite->AddData(xvalues, zhermiteYValues);
-      p_zhermite->AddCubicHermiteDeriv(zhermiteVelocities);
+      p_velocity[0] = state.velocity.x;
+      p_velocity[1] = state.velocity.y;
+      p_velocity[2] = state.velocity.z;
```

Comment:
Question for the future but since this is basically in ALE now do we even need to differentiate between the two states/sources?

Reply:
For interpolating the state, no. States::getState does hermite interpolation if velocities are available and linear otherwise. I'm not sure how it would impact the entire flow of SpicePosition to remove memCache.

Reply:
Fair, but since this "state" is being handled in ALE now it may be nice to reduce the overall complexity of the states in SpicePosition now. Not this sprint but it's a good half measure that someone could probably pick up at some other time